Mlem/Views/Shared/Posts/Post Sizes/Large Post.swift

```diff
@@ -23,66 +28,186 @@ struct LargePost: View {
 
     // parameters
     let postView: APIPostView
-    let isExpanded: Bool
+    
+    @Binding var layoutMode: LayoutMode
+    private var isExpanded: Bool {
+        layoutMode == .maximize
+    }
     
     // computed
-    var maxHeight: CGFloat { isExpanded ? .infinity : AppConstants.maxFeedPostHeight }
+    private var maxHeight: CGFloat {
+        switch layoutMode {
+        case .maximize:
+            return .infinity
+        case .preferredSize:
+            return AppConstants.maxFeedPostHeight
+        case .minimize:
+            return 44
+        }
+    }
     var titleColor: Color { !isExpanded && postView.read ? .secondary : .primary }
-
+    private var lineLimit: Int? {
+        switch layoutMode {
+        case .maximize:
+            return nil
+        case .preferredSize:
+            return 8
+        case .minimize:
+            return 2
+        }
```

Comment:
These can all be defined on the `LayoutMode` enum itself, then just used as `layoutMode.lineLimit` and `layoutMode.maxHeight`